tensorflow_probability/python/distributions/pert.py

```diff
@@ -0,0 +1,164 @@
+from tensorflow_probability.python.internal import prefer_static
+from tensorflow_probability.python.internal import tensor_util # pylint: disable=g-direct-tensorflow-import
+
+__all__ = [
+    'PERT',
+]
+
+class PERT(transformed_distribution.TransformedDistribution):
+  """Modified PERT distribution:
+  PERT distribution is a loc-scale family of Beta distribution intended to
+  model expert prediction on arbitrary real interval bounded by MAX and MIN
+  values set by the user, along with MODE to indicate the expert's
+  most frequent prediction.
+
+  The shift and scale factor is given by:
+
+  ```
+  shift = MIN
+  scale = MAX - MIN
+  ```
+
+  Modified PERT is a generalized version of PERT that has an additional
+  parameter, smoothness (g).
+
+  Smoothness determines the properties of Beta distribution:
+
+  ```
+  concentration1 = 1 + g * (MODE - MIN)/(MAX - MIN)
+  concentration0 = 1 + g * (MAX - MODE)/(MAX - MIN)
+  mean = (MIN + g * MODE + MAX)/(g + 2)
+  ```
+
+  PERT distribution is obtained when g = 4.
+
+  For general reference: https://en.wikipedia.org/wiki/PERT_distribution
+  """
+  def __init__(self,
+               mini,
+               mode,
+               maxi,
+               smoothness=4.,
+               validate_args=False,
+               allow_nan_stats=False,
+               name="PERT"):
+    parameters = dict(locals())
+    with tf.name_scope(name) as name:
+      dtype = dtype_util.common_dtype([mini, maxi, mode], tf.float32)
+      self._min = tensor_util.convert_nonref_to_tensor(
+          mini, name='mini', dtype=dtype)
+      self._mod = tensor_util.convert_nonref_to_tensor(
+          mode, name='mode', dtype=dtype)
+      self._max = tensor_util.convert_nonref_to_tensor(
+          maxi, name='maxi', dtype=dtype)
+      self._smoothness = tensor_util.convert_nonref_to_tensor(
+          smoothness, name='smoothness', dtype=dtype)
+      self._concentration1 = 1. + self._smoothness \
+        * (self._mod - self._min) / (self._max - self._min)
+      self._concentration0 = 1. + self._smoothness \
+        * (self._max - self._mod) / (self._max - self._min)
+
+      self._scale = self._max - self._min
+
+      # Broadcasting scale on affine bijector to match batch dimension.
+      # This prevents dimension mismatch for operations like cdf.
+      super(PERT, self).__init__(
+          distribution=Beta(
+              concentration1=self._concentration1,
+              concentration0=self._concentration0,
+              allow_nan_stats=allow_nan_stats),
+          bijector=AffineScalar(
+              shift=self._min,
+              scale=tf.broadcast_to(
+                  input=self._scale,
+                  shape=self._batch_shape_tensor(
+                      concentration1=self._concentration1,
+                      concentration0=self._concentration0))),
+          validate_args=validate_args,
+          parameters=parameters,
+          name=name)
+
```

Comment:
can you also implement _batch_shape (which is preferred when shapes are statically known)